appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy

```diff
@@ -12,168 +12,257 @@ import java.time.ZoneId
+
+	void setupTestScenario()
+	{
+		def richardAndBob = addRichardAndBobAsBuddies()
+		richard = richardAndBob.richard
+		bob = richardAndBob.bob
+		appService.addGoal(appService.&assertResponseStatusCreated, bob, TimeZoneGoal.createInstance(SOCIAL_ACT_CAT_URL, ["11:00-12:00"].toArray()), "Going to restrict my social time!")
+		def goals = appService.getGoals(bob)
+		bobBudgetGoalGamblingUrl = findGoal(goals, GAMBLING_ACT_CAT_URL)._links.self.href
+		bobBudgetGoalNewsUrl = findGoal(goals, NEWS_ACT_CAT_URL)._links.self.href
+		bobTimeZoneGoalUrl = findGoal(goals, SOCIAL_ACT_CAT_URL)._links.self.href
+
+		def richardWithBuddy = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateData, richard.url, true, richard.password)
+		assert richardWithBuddy.buddies != null
+		assert richardWithBuddy.buddies.size() == 1
+		def bobGoalsForRichard = richardWithBuddy.buddies[0].goals
+		bobBudgetGoalGamblingUrlForRichard = bobGoalsForRichard.find{it.activityCategoryUrl == GAMBLING_ACT_CAT_URL}.url
+		bobBudgetGoalNewsUrlForRichard = bobGoalsForRichard.find{it.activityCategoryUrl == NEWS_ACT_CAT_URL}.url
+		bobTimeZoneGoalUrlForRichard = bobGoalsForRichard.find{it.activityCategoryUrl == SOCIAL_ACT_CAT_URL}.url
+		bobDailyActivityReportsUrlForRichard = richardWithBuddy.buddies[0].dailyActivityReportsUrl
+		assert bobDailyActivityReportsUrlForRichard
+		bobWeeklyActivityReportsUrlForRichard = richardWithBuddy.buddies[0].weeklyActivityReportsUrl
+		assert bobWeeklyActivityReportsUrlForRichard
+
+		addSomeTestActivity(bob)
+		dateTime = ZonedDateTime.now(ZoneId.of("Europe/Amsterdam"))
+	}
+
+	void addSomeTestActivity(User user)
+	{
+		analysisService.postToAnalysisEngine(user, ["Gambling"], "http://www.poker.com")
+		analysisService.postToAnalysisEngine(user, ["social"], "http://www.facebook.com")
+	}
+
 	def 'Get day activity overviews'()
 	{
 		given:
-		def richard = addRichard()
-		def goals = appService.getGoals(richard)
-		def budgetGoalGamblingUrl = findGoal(goals, GAMBLING_ACT_CAT_URL)._links.self.href
-		def budgetGoalNewsUrl = findGoal(goals, NEWS_ACT_CAT_URL)._links.self.href
-		def timeZoneGoalUrl = appService.addGoal(appService.&assertResponseStatusCreated, richard, TimeZoneGoal.createInstance(SOCIAL_ACT_CAT_URL, ["11:00-12:00"].toArray()), "Going to restrict my social time!").url
-		analysisService.postToAnalysisEngine(richard, ["Gambling"], "http://www.poker.com")
-		analysisService.postToAnalysisEngine(richard, ["social"], "http://www.facebook.com")
+		setupTestScenario()
+
+		when:
+		def response = appService.getDayActivityOverviews(bob)
+
+		then:
+		testDayActivityOverviews(response, bobBudgetGoalGamblingUrl, bobBudgetGoalNewsUrl, bobTimeZoneGoalUrl)
+	}
+
+	def 'Get day activity overviews from buddy'()
+	{
+		given:
+		setupTestScenario()
 
 		when:
-		def response = appService.getDayActivityOverviews(richard)
+		def response = appService.getResourceWithPassword(bobDailyActivityReportsUrlForRichard, richard.password)
```

Comment:
Please move this into a method on ``AppService`` with this signature:
```
def getBuddyDayActivityOverviews(User user, int buddyIndex =  0, parameters = [:])
```
That allows doing this:
```
def response = appService.getBuddyDayActivityOverviews(richard)
```